docs/common-service-integration.md

```diff
@@ -0,0 +1,208 @@
+    packageName: jenkins-operator
+    sourceName: community-operators
+    sourceNamespace: openshift-marketplace
+    targetNamespaces:
+    - jenkins-operator
+...
+```
+
+`name` is the name of the operator, which should be same as the services name in the `CommonServiceConfig` and `CommonServiceSet`.
+`namespace` is the namespace the operator will be deployed in.
+`channel` is the name of tracked channel.
+`packageName` is the name of the package in `CatalogSource` will be deployed.
+`sourceName` is the name of the `CatalogSource`.
+`sourceNamespace` is the namespace of the `CatalogSource`.
+`targetNamespaces` is a list of namespace, which `OperaterGroup` generates RBAC access for its member Operators to get access to. `targetNamespaces` is used to control the operator dependency. `targetNamespaces` should include all the namespaces of its dependent operators and its own namespace.
+
+### Edit Common Service Config custom resource
+
+```bash
+cd common-service-operator
+vi deploy/crds/operator.ibm.com_v1alpha1_commonserviceconfig_cr.yaml
+```
+
+Append the operator custom resource information under `services` field.
+
+```yaml
+apiVersion: operator.ibm.com/v1alpha1
+kind: CommonServiceConfig
+metadata:
+  name: common-service
+spec:
+  services:
+  ...
+  - name: jenkins
+    spec:
+      Jenkins:
+        service:
+          port: 8081
+  ...
+```
+
+Take the jenkins operator as an example.
+The `name` field defines the name of the operator.
+The `spec` field defines the `spec` configuration for each custom resource in the service.
+
+In this example:
+The configuration for custom resource `Jenkins` is:
+
+```yaml
+      Jenkins:
+        service:
+          port: 8081
+```
+
+The configuration will be merged in the `spec` of the relevant `alm-example` in the `cluster service version` and generate custom resource `Jenkins`
+
+```yaml
+apiVersion: jenkins.io/v1alpha2
+kind: Jenkins
+metadata:
+  ...
+  name: example
+  namespace: jenkins-operator
+spec:
+  ...
+  service:
+    port: 8081
+    type: ClusterIP
+```
+
+### Edit Common Service Set custom resource
+
+```bash
+cd common-service-operator
+vi deploy/crds/operator.ibm.com_v1alpha1_commonserviceset_cr.yaml
+```
+
+Append the operator information under `services` field.
+
+```yaml
+apiVersion: operator.ibm.com/v1alpha1
+kind: CommonServiceSet
+metadata:
+  name: common-service
+spec:
+  services:
+  ...
+  - name: jenkins
+    channel: alpha
+    state: present
+    description: The jenkins service
+  ...
+```
+
+The `services` list defines the set for each service.
+`name` is the service name, which should be the same as the services name in the `CommonServiceConfig` and operator `name` in the `MetaOperator`.
+`channel` is an optional setting, it can overwrite the `channel` defined in the `MetaOperator`.
+`state` defines if the service should be present or absent.
+`description` is the description of the service.
+
+## 3.Make a pull request to merge the changes
+
+# End to end test
+
```

Comment:
I think we can add a note, before run the e2e test, use have to push your own CSV package to source